File: src/simulation/space/resource/dependency/TemperatureMin.java
package simulation.space.resource.dependency;

import simulation.space.Tile;
import simulation.space.resource.Resource;

public class TemperatureMin extends Temperature {
    public TemperatureMin(int threshold, double deprivationCoefficient) {
        super(threshold, deprivationCoefficient);
    }

    @Override
    public double satisfaction(Tile tile, Resource resource) {
        if (tile == null) {
            throw new RuntimeException();
        }
        double result = threshold -  tile.getTemperature();
        result = 1 / Math.sqrt(Math.max(0, result) + 1);
        return result;
    }

    @Override
    public boolean hasNeeded(Tile tile) {
        return tile.getTemperature() <= threshold;
    }
}

File: EcosystemSimulation/src/io/tashtabash/simulation/space/resource/dependency/TemperatureMin.kt
package io.tashtabash.simulation.space.resource.dependency

import io.tashtabash.simulation.space.resource.Resource
import io.tashtabash.simulation.space.tile.Tile
import kotlin.math.max
import kotlin.math.sqrt


class TemperatureMin(threshold: Int, deprivationCoefficient: Double) : Temperature(threshold, deprivationCoefficient) {
    override fun satisfaction(tile: Tile, resource: Resource): Double {
        var result = (threshold - tile.temperature).toDouble()
        result = 1 / sqrt(max(0.0, result) + 1)
        return result
    }

    override fun hasNeeded(tile: Tile) = tile.temperature <= threshold
}
